src/personaclient/PersonaClient.php

```diff
@@ -265,6 +273,36 @@ public function isPresignedUrlValid($url,$secret) {
         return $valid;
     }
 
+    /**
+     * Get a user profile based off a gupid passed in
+     * @param string $gupid
+     * @param string $token
+     * @access public
+     * @return mixed
+     * @throws \InvalidArgumentException
+     * @throws \Exception
+     */
+    public function getUserByGupid($gupid, $token){
```

Comment:
I thought that was only supported on a call to guids (i'm looking to add that as well so will definitely do it as part of this) - i'll check if its supported on gupid - if so i'll add it.